src/dataflow/concatenate_h5mu/script.py

```diff
@@ -220,6 +222,35 @@ def split_conflicts_modalities(
     return output
 
 
+def merge_obsp_block_diag(
+    mod_data: dict[str, anndata.AnnData],
+    concatenated_data: anndata.AnnData,
+) -> anndata.AnnData:
+    """
+    Build block-diagonal `.obsp` matrices for provided keys.
+    """
+
+    adatas_in_order = list(mod_data.values())
+
+    for key in par["obsp_keys"]:
+        # Evaluate that obsp key is present in all samples
+        if not all([key in adata.obsp.keys() for adata in adatas_in_order]):
+            logger.warning(
+                f".obsp['{key}'] not found in all samples, skipping block-diagonal merge for this key.",
+            )
+            continue
+
+        # Order of blocks must match the concat order
+        logger.info("Building block-diagonal obsp['%s'] matrix.", key)
+        blocks = []
+        for ad in adatas_in_order:
+            # `.tocsr()` to ensure compatible format for block_diag
+            blocks.append(ad.obsp[key].tocsr())
+        concatenated_data.obsp[key] = sp.block_diag(blocks, format="csr")
```

Comment:
`obsp` matrices have the following [types](https://github.com/scverse/anndata/blob/532386c38dd342170be6c97556b7a7ab0df49689/src/anndata/_core/aligned_mapping.py#L40C9-L40C55): `pd.DataFrame | CSMatrix | CSArray | np.ndarray`.
I'm not sure that this will work with all of those. Could you add a test for the different types?